### PR TITLE
feat(gem): add update & outdated subcommands

### PR DESCRIPTION
**Pull request by cadenkriese**
**What kind of change does this PR introduce?**
Adds completion spec for two subcommands of `gem`.

**What is the current behavior?**
Missing completion spec.

**What is the new behavior**
Implemented completion spec.

### CODE REVIEW

**Comment by SeparateRecords** [comment]
Hey @cadenkriese! Thanks for the PR 🙂

So this just adds the `outdated` and `update` subcommands but no options -- this is totally fine! Was wondering if you'd also be able to add their flags? No worries if you don't, we can absolutely merge as-is.

**Comment by SeparateRecords** [comment]
(also, https://github.com/withfig/autocomplete/pull/1555 is open, but this has a much smaller scope so can be merged sooner)

**Comment by cadenkriese** [comment]
No problem, I added the options. However, everything after `--prerelease` is duplicated between `gem install` and `gem update`. Should I put them in their own variable to share? Worth noting that I'm not familiar with typescript syntax, so pseudocode is much appreciated.

One other question, a lot of these options have an optional 'no-' prefix, ie. `--no-prerelease` to explicitly disable them, does fig have a convention for this?

File: src/gem.ts [hunk 1]
```diff
@@ -29,6 +29,180 @@ const completionSpec: Fig.Spec = {
...
+          args: {
+            name: "DIR",
+          },
+        },
+        {
+          name: "--document",
+          description: "Generate documentation for installed gems",
+          args: {
+            name: "TYPES",
+          },
```
Comment (SeparateRecords) [inline src/gem.ts:95]: Do we know ahead of time what types are available? If we do, let's add `suggestions`! It's an array of strings

File: src/gem.ts [hunk 2]
```diff
@@ -29,6 +29,180 @@ const completionSpec: Fig.Spec = {
...
+          name: "--post-install-message",
+          description: "Print post install message",
+        },
+        {
+          name: ["--file", "-g"],
+          description:
+            "Read from a gem dependencies API file and install the listed gems",
+          args: {
+            name: "FILE",
+          },
```
Comment (SeparateRecords) [inline src/gem.ts:176]: Can add `template: "filepaths"` for file suggestions 🙂

**Review by SeparateRecords — CHANGES_REQUESTED** [review]
> everything after --prerelease is duplicated between gem install and gem update. Should I put them in their own variable to share?

No need! I'm a big fan of WET (write everything twice), no need to simplify it yet. Can be refactored later to clean up 🙂

> Worth noting that I'm not familiar with typescript syntax, so pseudocode is much appreciated.

That's okay! Only thing you really need to know for Fig:
```
const [varname]: [type] = [essentially a json object]
```

For a TypeScript array, just add `[]` after the type name, so a list of Fig options is

```ts
const options: Fig.Option[] = [
  // objects go here
];
```

We've designed the API surface to be as minimal as possible so there's not much to learn, unless you want to get into some _really_ wild and wacky generators like [`deno doc`](https://github.com/withfig/autocomplete/blob/589d8d2d80fb301fb249573483179429941d5a39/src/deno/generators.ts#L7-L184) (provides real-time symbol suggestions)

https://user-images.githubusercontent.com/52195359/194490777-7f814728-59a0-4a61-b516-5f238d0dad45.mp4

> One other question, a lot of these options have an optional 'no-' prefix, ie. --no-prerelease to explicitly disable them, does fig have a convention for this?

There's no convention for this, you've done it perfectly!

File: src/gem.ts (hunk 1 above)
Comment (cadenkriese) [reply src/gem.ts]: I could only find the types 'rdoc' and 'ri' so I've added those. 👍

**Review by SeparateRecords — APPROVED** [review]
Looks good to me! Thanks @cadenkriese  🙂

**Pull request merged**